linchpin/provision/roles/azure/tasks/provision_azure_vm.yml

```diff
@@ -73,6 +86,7 @@
     admin_username: "{{ res_def['vm_username'] | default(linchpinUsername) }}"
     admin_password: "{{ res_def['vm_password'] | default(linchpinPassword) }}"
     resource_group: "{{ res_def['resource_group'] | default(omit) }}"
+    ssh_public_keys: "{{ssh_public_keys}}"
```

Comment:
how about  using `defaut(omit)` and also remove lines 75-79, will it work?

Reply:
No, it has to be NULL, it will not work if I use omit, Or I can write a filter which gives a NULL, but that way it will be inside py file,  this way it's clearer.

Reply:
then how about `default(NULL)`?

Reply:
I can't set a NULL directly to the default of the public key, because, even the public key is null, it's not gonna working. Ansible needs the field "ssh_public_keys" (contains "public key") to be NULL. Only set "public key" to null, the "ssh_public_keys" will be LIST("path/to/key",NULL) which is not NULL